### PR TITLE
[Noise] Adds errors for noiseless ops.

### PR DESCRIPTION
**Pull request by MichaelBroughton**
Adds error checking for all C++ ops that don't support noise.

### CODE REVIEW

File: tensorflow_quantum/core/ops/tfq_adj_grad_op_test.py [hunk 1]
```diff
@@ -223,6 +223,16 @@ def test_adj_grad_inputs(self):
                 tf.convert_to_tensor([[0, 0] for _ in range(len(circuit_batch))
                                      ]))
 
+        with self.assertRaisesRegex(tf.errors.InvalidArgumentError,
+                                    expected_regex='cirq.Channel'):
+            # Circuit tensor has too many dimensions.
```
Comment (jaeyoo) [inline tensorflow_quantum/core/ops/tfq_adj_grad_op_test.py:228]: nit: attempting to use noisy circuit.

File: tensorflow_quantum/core/ops/math_ops/inner_product_op_test.py [hunk 2]
```diff
@@ -190,6 +190,15 @@ def test_inner_product_inputs(self):
                 symbol_values_array,
                 util.convert_to_tensor([[x] for x in circuit_batch]))
 
+        with self.assertRaisesRegex(tf.errors.InvalidArgumentError,
+                                    expected_regex='cirq.Channel'):
+            # Circuit tensor has too many dimensions.
```
Comment (jaeyoo) [inline tensorflow_quantum/core/ops/math_ops/inner_product_op_test.py:195]: nit: attempting to use noisy circuit.

**Review by jaeyoo — COMMENTED** [review]
LGTM with nits

File: tensorflow_quantum/core/ops/tfq_adj_grad_op_test.py (hunk 1 above)
Comment (MichaelBroughton) [reply tensorflow_quantum/core/ops/tfq_adj_grad_op_test.py]: Good catch. Fixed.

File: tensorflow_quantum/core/ops/math_ops/inner_product_op_test.py (hunk 2 above)
Comment (MichaelBroughton) [reply tensorflow_quantum/core/ops/math_ops/inner_product_op_test.py]: Fixed.

**Review by jaeyoo — APPROVED** [review]

**Pull request merged**